### ISSUE TITLE
Stop using user's suffix as their last name

### ISSUE THREAD

**Issue by hschallhorn** [department-of-veterans-affairs/caseflow#12113]
As found in #11318, we store full names for users, some of which end in a suffix such as "JR" or "IV". With the introduction of #12107, we [sort on the back end](https://github.com/department-of-veterans-affairs/caseflow/blob/bd09097eddccba82a0a743318b82a767653e1bfc/app/models/task_sorter.rb#L75) the same way we [sort on the front end](https://github.com/department-of-veterans-affairs/caseflow/blob/bd09097eddccba82a0a743318b82a767653e1bfc/client/app/queue/components/TaskTableColumns.jsx#L243), by the [last full word](https://github.com/department-of-veterans-affairs/caseflow/blob/bd09097eddccba82a0a743318b82a767653e1bfc/app/models/serializers/work_queue/task_column_serializer.rb#L162) that appears in the [assigner's full name](https://github.com/department-of-veterans-affairs/caseflow/blob/bd09097eddccba82a0a743318b82a767653e1bfc/app/models/task.rb#L212). This leads to an assigned_by label of "Joe Jr" or sorting by "IV", rather than the person's real last name.

```ruby
> pp User.where('full_name ~* ?', ' (.*) ').group("substring(users.full_name,\'([a-zA-Z]+)$\')").order('count_full_name desc').order("substring(users.full_name,\'([a-zA-Z]+)$\') asc").count('full_name')
{
 "JR"=>27,
 "III"=>5,
 ...,
 "II"=>2,
 ...,
 "SR"=>2,
 ...,
 "I"=>1,
 "IV"=>1,
 ...,
 "V"=>1, # Neat!
 ...
}
```

## AC
- [ ] Front end displays assigner's true full name
- [ ] Front end sorts by an assigner's true last name
- [ ] Back end sorts by assigner's true last name

**Comment by hschallhorn** [comment]
TODO: Kick this to design to ask them why we don't use full names everywhere.

**Comment by hschallhorn** [comment]
@carodew or design in general, Any insight on this? 

We display assigner's name in queue as the first full word of their full name and the last full word of their full name (presumably to cut out their middle names). Is there any reason we don't just use their full name? A variation of this is shown in case details, where we only show the first letter of the assigner's first name, along with the last full word of their full name. 


<kbd><img width="166" alt="Screen Shot 2019-09-19 at 12 39 07 PM" src="https://user-images.githubusercontent.com/45575454/65263656-11a17e00-dadb-11e9-89c9-2dc48a4d75c0.png"></kbd><kbd><img width="268" alt="Screen Shot 2019-09-19 at 12 39 20 PM" src="https://user-images.githubusercontent.com/45575454/65263661-136b4180-dadb-11e9-9320-684861bf2b7b.png"></kbd>

The issue this causes is when there is a user with a suffix, "Joe Shmoe Jr" would be displayed as "Joe Jr" and "J. Jr"

**Comment by carodew** [comment]
Yeah that strikes me as a pretty peculiar way of displaying names. The only UI consideration I can think of is length of name vs. space in the interface. I imagine those questions were considered when these interface elements were designed, but I don't have the historical context to know what those decisions were based on. My gut is to agree – that we should specifically use the last name in the display, and not the last word. Do you have a sense for what the longest name is in the database (just character count), and how it would display in the interface?

As a related aside, there are a large number of edge cases to how people are named. We tend to think of a name as being comprised of title (optional), first name, middle name, surname, suffix (optional). But obviously other cultures have different naming conventions, such as surname first (China, probably others), two surnames (many Latinx communities), no middle name, or only one name. I'm sure there are variations I'm not considering here as well. So creating general rules for how to handle people's names are bound to be fraught. That said, since we're pulling that data from another system presumably they've made decisions on how to handle those edge cases and have already applied some type of normative data handling, so I'm guessing we don't really have wiggle room or authority to maneuver in this area. Mostly just providing some additional context to think about in case we make a change and start seeing some funny business afterwards.

**Comment by carodew** [comment]
Oh, also for relatively brief questions (Drive-by Opinions™, as it were) I'm happy to just be pinged and comment without moving pipelines, but didn't realize that the question in Slack was a relatively small ask. But don't worry too much about whether or not a thing should be moved, we can always fix it later.

**Comment by hschallhorn** [comment]
30 characters is the longest full name in our system, which is "DONOTDELETE CASEFLOWSYSTEMACCT".  The second longest (and an actual person) is 28 characters.

Naming conventions were also brought up when we talked about this ticket, which obviously gave us more reason to want to display full names.

**Comment by hschallhorn** [comment]
Annnnnd how this would look in these two instances
<img width="1094" alt="Screen Shot 2019-09-19 at 3 21 36 PM" src="https://user-images.githubusercontent.com/45575454/65274424-40c2ea00-daf1-11e9-9e4e-23d18d65b439.png">
<img width="955" alt="Screen Shot 2019-09-19 at 3 21 47 PM" src="https://user-images.githubusercontent.com/45575454/65274425-428cad80-daf1-11e9-8063-d6e9477e663d.png">

**Comment by carodew** [comment]
Basically it just wraps, which I think is fine for most instances. 

I tested a 30 character name in the `assign to` dropdown in Queue, which got a little weird at narrower widths:

![image](https://user-images.githubusercontent.com/10144074/65274713-fd1cb000-daf1-11e9-878b-93ab679bc466.png)

In most other places, it seems to just wrap. I haven't checked on all the places it could appear in the UI. Would changing this behavior impact all the places a name would appear in Caseflow or just the two screenshots you've shown above? If the former, maybe propose the change in a Appeals-wide channel to see if other people know if places in the UI where changing to a longer name might impact? I don't offhand know of all the possible instances myself. 

related, I wonder if we should be sure to have an example of at least a 30-character name in Caseflow demo so that we can always see what that might look like in the UI? (probably both for veterans and for other user types). IDK how hard that would be so just throwing it out there.

**Comment by pkarman** [comment]
FWIW I believe the SAML payload we get on PIV log in has separate fields for first and last name, and we just happen to be truncating them. If it would help, we could investigate what it would take to actually store the separate name fields in our User table to make this simpler.

**Comment by carodew** [comment]
Yeah, IDK how hard it would be and it does seem like a small change, but it may also be worthwhile since 1) it seems like the way names are currently displayed could create confusion while assigning things, which could have small but existent impacts on timeliness, and 2) people's sense of identity are closely tied up with their names and therefore emotions can run strong when their name is wrong, which I imagine could be exacerbated when you work in a large bureaucracy.
